frontend/src/app/team.service.mock.ts

```diff
@@ -14,15 +16,41 @@ export class TeamServiceMock {
     },
   ];
 
-  getTeams(): Observable<Team[]> {
-    return of({status: 'success', data: this.teams} as any);
+  teamsSubject = new BehaviorSubject<QueryOutput<Team[]>>({status: 'success', data: this.teams} as any);
+
+  getTeams(): Observable<QueryOutput<Team[]>> {
+    return this.teamsSubject.asObservable();
   }
 
-  getTeamById(id: number): Observable<Team> {
-    return of(this.teams.find(team => team.id === id)!);
+  getTeamById(id: number): Observable<QueryOutput<Team>> {
+    return of({status: 'success', data: this.teams.find(team => team.id === id)!} as any);
   }
 
   updateTeam(team: Team): Observable<Team> {
-    return of(team);
+    const foundTeam = this.teams.find(existingTeam => existingTeam.id === team.id)!;
+    updateAllAttributes(team, foundTeam);
+    this.teamsSubject.next({status: 'success', data: this.teams} as any);
+    return of(foundTeam);
+  }
+
+  createTeam(): Observable<Team> {
+    const newTeam = {
+      id: Math.floor(Math.random() * 1000) + 1,
+      name: 'Uusi tiimi',
+      choreographies: [],
+      users: [],
+      people: [],
+    };
+    this.teams.push(newTeam);
+    this.teamsSubject.next({status: 'success', data: this.teams} as any);
+    return of(newTeam);
+  }
+
+  deleteTeamById(id: number): Observable<Team> {
+    const foundTeam = this.teams.find(existingTeam => existingTeam.id === id)!;
```

Comment:
**chore:** delete this one as the method should return just `Observable<void>` instead of `Observable<Team>`